lib/state_notifiers/push_request_notifier.dart

```diff
@@ -33,16 +34,18 @@ import 'package:privacyidea_authenticator/utils/push_provider.dart';
 import 'package:privacyidea_authenticator/utils/riverpod_providers.dart';
 import 'package:privacyidea_authenticator/utils/storage_utils.dart';
 
+import '../utils/customizations.dart';
+
 class PushRequestNotifier extends StateNotifier<PushRequest?> {
   // Used for periodically polling for push challenges
-  bool pollingEnabled;
   static Timer? _pollTimer;
+  final bool pollingEnabled;
 
   PushRequestNotifier(super.state, {required this.pollingEnabled}) {
     _initStateAsync();
   }
 
-  void _startPollingIfEnabled() {
+  void _startOrStopPolling() {
```

Comment:
_pollTimer calls the poll method every period until it's canceled.
because of pollingEnabled PushRequestNotifier has to be based on the settingsState wich contains pollingEnabled.
When there is a new settingsState a new PushRequestNotifier is created
in _initStateAsync the private method _startOrStopPolling is called:
when pollingEnabled is false and there is a _pollTimer then cancel the timer and set the _pollTimer to null